File: chatdownloader/src/leaderboards/overall.py
"""
The overall leaderboard
"""

import json
import logging
import os
from dataclasses import dataclass

from _types import LeaderboardExportItem, UserChatPerformance

from .abstractleaderboard import AbstractLeaderboard

DEFAULT_ELO = 1200
K = 0.5


class Overall(AbstractLeaderboard):
    """
    Computes the overall leaderboard (Special + Non-Special).
    """
    def __init__(self):
        self.state = {}
        super().__init__()

    @classmethod
    def get_name(cls):
        return 'overall'

    def calculate_score(self, performance: UserChatPerformance) -> float:
        return sum(performance.metrics.values())

File: chatdownloader/src/leaderboards/overall.rs
/*
The overall leaderboard
*/

use crate::leaderboards::leaderboardtrait::AbstractLeaderboard;
use crate::_types::clptypes::UserChatPerformance;
use crate::_types::leaderboardtypes::LeaderboardInnerState;
use std::collections::HashMap;

pub struct ChatOnly {
    state: HashMap<String, LeaderboardInnerState>,
}

impl AbstractLeaderboard for ChatOnly {
    fn new() -> Self {
        Self {
            state: HashMap::new(),
        }
    }

    fn get_name(&self) -> &str {
        "overall"
    }

    fn __get_state(&mut self) -> &mut HashMap<String, LeaderboardInnerState> {
        &mut self.state
    }

    fn calculate_score(&self, performance: &UserChatPerformance) -> Option<f32> {
        return Some(performance.metrics.values().sum());
    }
}